### PR TITLE
BUGZ-838, BUGZ-850, BUGZ-853: Muted sign displays backwards in second person view, Fix lossy/duplicate color format conversions, Adjust wearables screen is flipped

### PR DESCRIPTION
**Pull request by annabrewer**
https://highfidelity.atlassian.net/browse/BUGZ-838
https://highfidelity.atlassian.net/browse/BUGZ-850
https://highfidelity.atlassian.net/browse/BUGZ-853

Simplified the logic in our tone mapping and display plugin files to remove an extra conversion from RGB to SRGB and back to RGB. Did this by removing the conversion from RGB to SRGB in each of our tone mapping curves, since that conversion was already being done due to the call to glEnable(GL_FRAMEBUFFER_SRGB) in GLBackendOutput. The composite framebuffer is now SRGB.

Also fixed the issue where the UI was rendering backwards in mirror / second person mode. This happened because the HUD pipeline in the OpenGL display plugin was undoing a horizontal flip of the UI (as part of the composited frame) that was supposed to happen in Blit::run in RenderCommonTask, but didn't. I've moved the mirroring to UpsampleToBlitFramebuffer::run in ResampleTask, so the UI never gets mirrored in the first place.

### CODE REVIEW

File: libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp [hunk 1]
```diff
@@ -582,17 +579,16 @@ void OpenGLDisplayPlugin::updateFrameData() {
...
     auto hudStereo = isStereo();
     auto hudCompositeFramebufferSize = _compositeFramebuffer->getSize();
     std::array<glm::ivec4, 2> hudEyeViewports;
     for_each_eye([&](Eye eye) {
         hudEyeViewports[eye] = eyeViewport(eye);
     });
-    return [=](gpu::Batch& batch, const gpu::TexturePointer& hudTexture, bool mirror) {
-        auto pipeline = mirror ? hudMirrorPipeline : hudPipeline;
+    return [=](gpu::Batch& batch, const gpu::TexturePointer& hudTexture) {
+        auto pipeline = hudPipeline;
```
Comment (SamGondelman) [inline libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: you can get rid of this and replace `pipeline` on the lines below here with `hudPipeline`

File: libraries/render-utils/src/toneMapping.slf [hunk 2]
```diff
@@ -50,14 +51,24 @@ void main(void) {
     int toneCurve = getToneCurve();
     vec3 tonedColor = srcColor;
     if (toneCurve == ToneCurveFilmic) {
-        vec3 x = max(vec3(0.0), srcColor-0.004);
+        vec3 rgbColor = pow(srcColor, vec3(GAMMA_22));
```
Comment (SamGondelman) [inline libraries/render-utils/src/toneMapping.slf]: why this extra conversion here?

File: libraries/render-utils/src/toneMapping.slf [hunk 3]
```diff
@@ -50,14 +51,24 @@ void main(void) {
...
+        vec3 x = max(vec3(0.0), rgbColor-0.004);
         tonedColor = (x * (6.2 * x + 0.5)) / (x * (6.2 * x + 1.7) + 0.06);
     } else if (toneCurve == ToneCurveReinhard) {
         tonedColor = srcColor/(1.0 + srcColor);
-        tonedColor = pow(tonedColor, vec3(INV_GAMMA_22));
-    } else if (toneCurve == ToneCurveGamma22) {
-        tonedColor = pow(srcColor, vec3(INV_GAMMA_22));
-    } // else None toned = src
+    } 
+    else if (toneCurve == ToneCurveGamma22) {
```
Comment (SamGondelman) [inline libraries/render-utils/src/toneMapping.slf]: These `else if`s should be on the same line as the `}`

File: libraries/render/src/render/ResampleTask.cpp [hunk 4]
```diff
@@ -164,7 +169,7 @@ void UpsampleToBlitFramebuffer::run(const RenderContextPointer& renderContext, c
             batch.setViewportTransform(viewport);
             batch.setProjectionTransform(glm::mat4());
             batch.resetViewTransform();
-            batch.setPipeline(_pipeline);
+            batch.setPipeline(pipeline);
```
Comment (SamGondelman) [inline libraries/render/src/render/ResampleTask.cpp]:
```suggestion
            batch.setPipeline(args->_renderMode == RenderArgs::MIRROR_RENDER_MODE ? _mirrorPipeline : _pipeline);
```
(this avoids the extra copy when you create `pipeline` above

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20201/HighFidelity-Beta-PR15862-95f2dfa.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20201/HighFidelity-Quest-Beta-PR15862-95f2dfa.apk)

File: libraries/render-utils/src/toneMapping.slf (hunk 2 above)
Comment (annabrewer) [reply libraries/render-utils/src/toneMapping.slf]: because it was too light before, needed to go back to rgb before applying the filmic formula

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20201/HighFidelity-Beta-PR15862-95f2dfa.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20201/HighFidelity-Beta-Interface-PR15862-95f2dfa.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20201/HighFidelity-Beta-PR15862-95f2dfa.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20201/HighFidelity-Beta-Interface-PR15862-95f2dfa.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20204/HighFidelity-Beta-PR15862-f44bc52.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20204/HighFidelity-Beta-Interface-PR15862-f44bc52.dmg

**Comment by annabrewer** [comment]
Tested on all devices, looks good to go!

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20204/HighFidelity-Beta-PR15862-f44bc52.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20204/HighFidelity-Beta-Interface-PR15862-f44bc52.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20204/HighFidelity-Beta-PR15862-f44bc52.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20204/HighFidelity-Quest-Beta-PR15862-f44bc52.apk)

**Pull request merged**